File: 7-kyu/holiday-ii---plane-seating/python/solution.py
def plane_seat(a):
    seat_section = int(a[:-1])
    seat_cluster = a[-1]
    if seat_section > 60 or seat_cluster not in 'ABCDEFGHK':
        return 'No Seat!!'
    ret_section = 'Front' if seat_section <= 20 else 'Middle' if seat_section <= 40 else 'Back'
    ret_cluster = 'Left' if seat_cluster in 'ABC' else 'Middle' if seat_cluster in 'DEF' else 'Right'
    return f'{ret_section}-{ret_cluster}'

File: 7-kyu/holiday-ii---plane-seating/javascript/solution.js
const planeSeat = (a) => {
  if (/[IJ]/.test(a) || a.length === 3 && parseInt(a.slice(0, 2)) > 60) {
    return "No Seat!!";
  }
  const section = a.length === 3 ? parseInt(a.slice(0, 2)) : parseInt(a.slice(0, 1));
  const cluster = a.at(-1);
  let sectionLocation = "";
  let clusterLocation = "";
  if (section < 21) {
    sectionLocation = "Front";
  } else if (section < 41) {
    sectionLocation = "Middle";
  } else {
    sectionLocation = "Back";
  }
  if (cluster >= "A" && cluster <= "C") {
    clusterLocation = "Left";
  } else if (cluster >= "D" && cluster <= "F") {
    clusterLocation = "Middle";
  } else {
    clusterLocation = "Right";
  }
  return `${sectionLocation}-${clusterLocation}`;
};
